lib/url.js

```diff
@@ -250,10 +250,11 @@ Url.prototype.parse = function parse(url, parseQueryString, slashesDenoteHost) {
     }
   }
 
-  var proto = protocolPattern.exec(rest);
+  let proto = protocolPattern.exec(rest);
+  let lowerProto;
   if (proto) {
     proto = proto[0];
-    var lowerProto = proto.toLowerCase();
+    lowerProto = proto.toLowerCase();
     this.protocol = lowerProto;
```

Comment:
I suggest to just directly set this to `proto.toLowerCase()` instead of using an intermediate variable.

Reply:
That's my first thought too. But it's actually used by 
https://github.com/nodejs/node/blob/f8763bb077db2f0be74e353c0b4f9e353c0fffa8/lib/url.js#L401
and other places.

That's also why I have to move the declaration to upper scope.